tests/test_app.py

```diff
@@ -38,16 +37,17 @@ def get_num_rows() -> dict[str, int]:
     if write_to_db:
         cmd_args.append("--write-to-db")
 
-    result = runner.invoke(main, cmd_args, catch_exceptions=False)
-
-    if result.exit_code != 0:
-        traceback.print_exception(result.exception)
-
-    assert result.exit_code == 0
+    result = runner.invoke(main, cmd_args, catch_exceptions=True)
 
     # Without this the output to stdout/stderr is grabbed by click's test runner.
     print(result.output)
 
+    # In case of an exception, raise it so that the test fails with the exception.
```

Comment:
nice, i like it

Reply:
Yeah it took many iterations to get to this setup!